### PR TITLE
[JENKINS-70379] upgrade to run on java 11

### PR DESCRIPTION
**Pull request by car-roll**
# Description

See [JENKINS-70379](https://issues.jenkins.io/browse/JENKINS-70379).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitReadSaveService.java [hunk 1]
```diff
@@ -68,6 +69,7 @@
 public class GitReadSaveService extends ScmContentProvider {
 
     private static final Logger logger = LoggerFactory.getLogger(GitReadSaveService.class);
+    @SuppressFBWarnings(value={"NP_NONNULL_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR"}, justification="https://github.com/spotbugs/spotbugs/issues/2044")
```
Comment (car-roll) [inline blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitReadSaveService.java:72]: not really sure if this is the issue. It seems like if it really was the issue it wouldn't have been ignored for so long...most likely something i need to fix (environment?). Wanted to quickly get past this error for now to do more testing.

File: pom.xml [hunk 2]
```diff
@@ -28,12 +28,11 @@
     <revision>1.26.1</revision>
     <changelist>-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/blueocean-plugin</gitHubRepo>
-    <java.level>8</java.level>
     <!--
         when changing jenkins core version please remember to change it in Jenkinsfile as well jenkinsVersions
         and  acceptance-tests/runner/scripts/args.sh
```
Comment (jglick) [inline pom.xml:33]: ^^^

File: blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitBareRepoReadSaveRequest.java [hunk 3]
```diff
@@ -164,6 +165,7 @@ <T> T invokeOnScm(final GitSCMFileSystem.FSFunction<T> function) throws IOExcept
         }
     }
 
+    @SuppressFBWarnings(value = "DCN_NULLPOINTER_EXCEPTION", justification = "Empty repository throws NPE")
```
Comment (jglick) [inline blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitBareRepoReadSaveRequest.java:168]:
```suggestion
    @SuppressFBWarnings(value = "DCN_NULLPOINTER_EXCEPTION", justification = "TODO check if this NPE is still real, and if so fix it")
```

File: blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitReadSaveService.java (hunk 1 above)
Comment (jglick) [reply blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitReadSaveService.java:72]: https://github.com/spotbugs/spotbugs/issues/2044 FTR

```suggestion
    @SuppressFBWarnings(value={"NP_NONNULL_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR"}, justification="https://github.com/spotbugs/spotbugs/issues/2044 TYPE is mutated by setType")
```

File: blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitUtils.java [hunk 4]
```diff
@@ -48,8 +48,8 @@
 import org.eclipse.jgit.revwalk.RevCommit;
 import org.eclipse.jgit.revwalk.RevTree;
 import org.eclipse.jgit.revwalk.RevWalk;
-import org.eclipse.jgit.transport.JschConfigSessionFactory;
-import org.eclipse.jgit.transport.OpenSshConfig;
+import org.eclipse.jgit.transport.ssh.jsch.JschConfigSessionFactory;
```
Comment (jglick) [inline blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitUtils.java:51]: https://issues.jenkins.io/browse/JENKINS-70379 FTR

File: pom.xml [hunk 5]
```diff
@@ -663,13 +662,18 @@
...
         </dependency>
         <dependency>
             <groupId>com.google.errorprone</groupId>
             <artifactId>error_prone_annotations</artifactId>
             <version>2.5.1</version>
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.modules</groupId>
+            <artifactId>instance-identity</artifactId>
+            <version>116.vf8f487400980</version>
```
Comment (jglick) [inline pom.xml:675]: https://github.com/jenkinsci/bom/issues/1647

File: pom.xml (hunk 2 above)
Comment (car-roll) [reply pom.xml:33]: 🤦

File: blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitReadSaveService.java (hunk 1 above)
Comment (car-roll) [reply blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitReadSaveService.java]: resolved in e4f44aba58fa7baf959f7a5b30aad54f599c566e

File: blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitBareRepoReadSaveRequest.java (hunk 3 above)
Comment (car-roll) [reply blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitBareRepoReadSaveRequest.java]: resolved in e4f44aba58fa7baf959f7a5b30aad54f599c566e

File: blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitUtils.java (hunk 4 above)
Comment (car-roll) [reply blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitUtils.java:51]: updated PR title

**Comment by car-roll** [comment]
Would the continuous-integration build failure stop incrementals from being built?

**Comment by MarkEWaite** [comment]
> Would the continuous-integration build failure stop incrementals from being built?

The failure is on the outdated blue ocean specific CI server that the infra team is decommissioning.  https://ci.jenkins.io/job/Plugins/job/blueocean-plugin/view/change-requests/job/PR-2379/lastSuccessfulBuild/artifact/ shows a successful build and a zip of artifacts to download.  I'm going to try that zip of artifacts in my Jenkins installation as an experiment.

**Comment by jglick** [comment]
https://repo.jenkins-ci.org/incrementals/io/jenkins/blueocean/blueocean/1.26.1-rc4327.e4f44aba58fa/ (and other artifacts) got deployed OK.

**Review by jglick — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]
I deployed 1.26.1-rc4327.e4f44aba58fa and confirmed that the scenario described in [JENKINS-70379](https://issues.jenkins.io/browse/JENKINS-70379) works with Jenkins 2.375.1.

The code changes look very good to me as well.

**Comment by jglick** [comment]
:shipit: @olamy @dwnusbaum @batmat

**Comment by shamil** [comment]
any reason I see this change in the update manager?

image

**Comment by car-roll** [comment]
@shamil thanks for catching that, will fix

**Review by dwnusbaum — APPROVED** [review]

**Comment by olamy** [comment]
> > Would the continuous-integration build failure stop incrementals from being built?
> 
> 
> 
> The failure is on the outdated blue ocean specific CI server that the infra team is decommissioning.  https://ci.jenkins.io/job/Plugins/job/blueocean-plugin/view/change-requests/job/PR-2379/lastSuccessfulBuild/artifact/ shows a successful build and a zip of artifacts to download.  I'm going to try that zip of artifacts in my Jenkins installation as an experiment.
> 

Not exactly. Failures are happening in ATH test module which is not running per default on ci.j.io 

> 
> 
> I don't know if Blue Ocean builds are equipped to deliver incremental.

**Pull request merged**